detox/e2e/test/messaging/message_posting.e2e.js

```diff
@@ -26,4 +29,23 @@ describe('Messaging', () => {
 
         await expect(element(by.text(text))).toExist();
     });
+
+    it('MM-T3187 Unread channels sort at top', async () => {
+        // # Open channel drawer (with at least one unread channel)
+        await element(by.id('channel_drawer_button')).tap();
+
+        // # Verify unread channel(s) display at top of channel list (with mentions first, if any), in alphabetical order, with title "Unreads"
+        await expect(element(by.text('UNREADS'))).toBeVisible();
```

Comment:
I'm not sure how to ensure the channels are in alphabetical order.

Reply:
Hi @clarmso I tried adding `testID='channel_item_display_name'` to the ChannelItem > render > `Text` element of `app/components/sidebars/main/channel_list/channel_item/channel_item.js`

and then added this verification, which works,
```
await expect(element(by.id('channel_item_display_name')).atIndex(0)).toHaveText(newChannel.display_name);
```

You can probably add two channels, channel1 = 'B Channel', channel2 = 'A Channel', then have the following verification for order,

```
await expect(element(by.id('channel_item_display_name')).atIndex(0)).toHaveText(channel2.display_name);
await expect(element(by.id('channel_item_display_name')).atIndex(1)).toHaveText(channel1.display_name);
```

Reply:
If multiple elements on the same screen can have the same ID, I'm ok. (For web, it's not a good thing to have the same ID used in multiple elements.) 

I'll update the snapshot if necessary. :)

Reply:
You can construct the id to be something like `channel_item.${channel.id}` to avoid repetition if the same id

Reply:
@enahum Good point. If unique ID isn't required for mobile, I'd leave it as is for now.

Reply:
In addition, the "Unreads" should be at the top. I think this heading is displayed using `/app/components/sidebars/main/channels_list/filtered_list/filtered_list.js`. In particular, "Unreads" may be rendered via `renderSectionHeader`.  I have tried adding `testID` under `<View>` and `<Text>` but have not gotten the desired effect.

Verifying the "Unreads" exists is weaker than verifying it's the first heading. :(